File: Node/model/block.js
/**
 * Created by milenradkov on 2/2/18.
 */
export class Block {
    constructor(index, transactions, difficulty, prevBlockHash, minedBy, blockDataHash, nonce, dateCreated, blockHash) {
        // Index: number
        this.index = index;

        // Transactions: Transaction[]
        this.transactions = transactions;

        // PrevBlockHash: hex_number
        this.prevBlockHash = prevBlockHash.toString();

        // BlockDataHash: address
        this.blockDataHash = blockDataHash.toString();

        // MinedBy: address
        this.minedBy = minedBy;

        // Nonce: number
        this.nonce = nonce;

        // Difficulty: number
        this.difficulty = difficulty;

        // DateCreated: timestamp
        this.dateCreated = dateCreated;

        // BlockHash: hex_number
        this.blockHash = blockHash;
    }
}

File: NodeDotNet/NodeDotNet.Core/Models/Block.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace NodeDotNet.Core.Models
{
    public class Block
    {
        public int Index { get; set; }
        public List<Transaction> Transactions { get; set; }
        public long Difficulty { get; set; }
        public string PrevBlockHash { get; set; }
        public Address MinedBy { get; set; }
        public string BlockDataHash { get; set; }
        public long Nonce { get; set; }
        public DateTime CreatedOn { get; set; }
        public string BlockHash { get; set; }

        public void GenerateHashes()
        {
            this.BlockDataHash = "hardcoded data hash, TODO merkel tree and hashing";
            this.CreatedOn = DateTime.UtcNow;
            this.Nonce = 1234567890;
            this.BlockHash = "hardcoded block hash - TODO use mining process";
        }

        private static Block _genesisBlock;
        public static Block Genesis
        {
            get
            {
                if (_genesisBlock == null)
                {

                    _genesisBlock = new Block
                    {
                        Index = 0,
                        Transactions = new List<Transaction>()
                        {
                            new Transaction{
                                From = Address.GeneratorAddress,
                                Amount = 100,
                                To = new Address
                                {
                                    AddressId = "1"
                                },
                                ReceivedOn = DateTime.UtcNow,
                                SenderPublickKey = "hardocoded SenderPublicKey",
                                SenderSignature = new List<string> {"hardcoded SenderSignatoure", "somee" }
                            },
                        },
                        Difficulty = 1,
                        MinedBy = new Address
                        {
                            AddressId = "00",
                            Amount = 10000
                        },
                        PrevBlockHash = String.Empty
                    };
                    _genesisBlock.GenerateHashes();
                }

                return _genesisBlock;
            }
        }
    }
}
